packages/sanity/src/desk/panes/document/documentPanel/DeletedDocumentBanner.tsx

```diff
@@ -1,46 +1,49 @@
-import React, {useCallback, useMemo} from 'react'
+import React, {useCallback} from 'react'
 import {Button, Card, Container, Flex, Text} from '@sanity/ui'
 import {ReadOnlyIcon} from '@sanity/icons'
 import styled from 'styled-components'
 import {useDocumentPane} from '../useDocumentPane'
-import {useDocumentOperation, useTimelineSelector} from 'sanity'
+import {useDocumentOperation} from 'sanity'
+import {useRouter} from 'sanity/router'
 
 const Root = styled(Card)`
   position: relative;
   z-index: 50;
 `
-export function DeletedDocumentBanner() {
-  const {documentId, documentType, timelineStore} = useDocumentPane()
-  const chunks = useTimelineSelector(timelineStore, (state) => state.chunks)
-  const lastRevisionId = useMemo(() => {
-    const lastRevision = chunks.filter((chunk) => !['delete', 'initial'].includes(chunk.type))
 
-    return lastRevision?.[0]?.id
-  }, [chunks])
```

Comment:
**suggestion:** this was removed in favour of creating a targeted selector that only returns the last non-deleted revision ID. 

Whilst not super applicable to this banner (as it's only visible when the document is deleted), selecting `chunks` can cause a bunch of unnecessary renders as it updates very frequently